gotham/src/helpers/http/response/mod.rs

```diff
@@ -64,10 +64,22 @@ type Body = (Vec<u8>, Mime);
 /// #     );
 /// # }
 /// ```
-pub fn create_response(state: &State, status: StatusCode, body: Option<Body>) -> Response {
-    let mut res = Response::new();
-    extend_response(state, &mut res, status, body);
-    res
+pub fn create_response(
+    state: &State,
+    status: StatusCode,
+    body: Option<DataMime>,
+) -> Response<Body> {
+    let mut builder = Response::builder();
+    let mime = body.clone().map(|(_, mime)| mime);
+
+    extend_response(state, status, &mut builder, mime);
+    match body {
+        Some((body, _)) => match *Method::borrow_from(state) {
+            Method::HEAD => builder.body(Body::empty()),
+            _ => builder.body(body.into()),
+        },
+        None => builder.body(Body::empty()),
+    }.unwrap()
```

Comment:
It's unfortunate that we have to stay compatible here, because it would be nice to return a `Result` rather than unwrapping here.

Reply:
Having said that, if this is `0.3.0`, maybe we can change this?

Reply:
I'd be happy to see that tackled in a separate PR. `create_response` gets called in a bunch of places, though.